File: spotifyApp/src/app/model/Track.ts
import { Album } from './Album';
import { Artist } from './Artist';
import { ExternalUrls } from './ExternalUrls';

export class Track{
    album!: Album;
    artists!: Artist[];
    external_urls!: ExternalUrls;
    id!: string;
    name!: string;
    preview_url!: string;
    type!: string;
    duration_ms!: number;
    index!: number;
    // constructor(
    //     public album: Album,
    //     public artists: Artist[],
    //     public external_urls: ExternalUrls,
    //     public id: string,
    //     public name: string,
    //     public preview_url: string,
    //     public type: string,
    //     public duration_ms: number
    // ) {}
    
}

File: temp/backend/wishlist-service/src/main/java/com/spotify/wishlistservice/model/Track.java
package com.spotify.wishlistservice.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document
public class Track {

    @JsonProperty("album")
    private Album album;

    @JsonProperty("artists")
    private List<Artist> artists;

    @JsonProperty("external_urls")
    private ExternalUrls externalUrls;

    @JsonProperty("href")
    private String href;

    @Id
    @JsonProperty("id")
    private String id;

    @JsonProperty("name")
    private String name;

    @JsonProperty("preview_url")
    private String previewUrl;

    @JsonProperty("type")
    private String type;

    @JsonProperty("uri")
    private String uri;

    // Getters and setters
}
